score.go

```diff
@@ -334,6 +376,45 @@ func (ps *peerScore) inspectScores() {
 	go ps.inspect(scores)
 }
 
+func (ps *peerScore) inspectScoresExtended() {
+	ps.Lock()
+	scores := make(map[peer.ID]*PeerScoreSnapshot, len(ps.peerStats))
+	for p, pstats := range ps.peerStats {
+		pss := new(PeerScoreSnapshot)
+		pss.Score = ps.score(p)
+		if len(pstats.topics) > 0 {
+			pss.Topics = make(map[string]*TopicScoreSnapshot, len(pstats.topics))
+			for t, ts := range pstats.topics {
+				pss.Topics[t] = &TopicScoreSnapshot{
+					FirstMessageDeliveries:   ts.firstMessageDeliveries,
+					MeshMessageDeliveries:    ts.meshMessageDeliveries,
+					InvalidMessageDeliveries: ts.invalidMessageDeliveries,
+				}
+				if ts.inMesh {
+					pss.Topics[t].TimeInMesh = ts.meshTime
```

Comment:
nit: instantiate the `TopicScoreSnapshot` as a local var, populate it, then set it _once_ in the map. Maybe the compiler optimises this double access, but I'm not sure, and this will be called quite frequently.

Reply:
sure, will do.

Reply:
done.